tests/integration/states/archive.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+'''
+Tests for the file state
+'''
+# Import python libs
+from __future__ import absolute_import
+import os
+import shutil
+
+# Import Salt Testing libs
+from salttesting.helpers import ensure_in_syspath
+ensure_in_syspath('../../')
+
+# Import salt libs
+import integration
+import salt.utils
+
+STATE_DIR = os.path.join(integration.FILES, 'file', 'base')
+ARCHIVE_DIR = '/tmp/archive/'
+ARCHIVE_TAR_SOURCE = 'https://github.com/downloads/Graylog2/'\
```

Comment:
You are right you can use `salt://` with archive, but when I went to change the test to using `salt:// `instead of `http` I found out that we do not check the `source_hash` on purpose if the file is on the salt master [here](https://github.com/saltstack/salt/blob/2016.3/salt/modules/file.py#L4436) , which makes the test to ensure the source_hash is checked correctly obsolete. So I do not see a way around it for that test. I can add  a tar.gz to the test suite for the other test and any other tests added in the future. are you okay with that? Or we could add an argument to check for source on the salt master in the code.